### PR TITLE
PP-2756 Add notifySettings to gatewayAccountEntity

### PR DESCRIPTION
**Pull request by SandorArpa**
- add a new json type field to gatewayAccountEntity to store notify API settings for custom branding
- refactor CredentialsConverter to be a generic JSON type converter 
- adding unit and integration tests for the entity and the converter

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/model/domain/JsonDBFieldConverter.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import javax.persistence.AttributeConverter;
+import javax.persistence.Converter;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Map;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+@Converter
+public class JsonDBFieldConverter implements AttributeConverter<Map<String, String>, PGobject> {
```
Comment (kakumara) [inline src/main/java/uk/gov/pay/connector/model/domain/JsonDBFieldConverter.java]: 👍 nice. How about a name like `JsonToMapConverter` ?

File: src/test/java/uk/gov/pay/connector/it/dao/GatewayAccountDaoITest.java [hunk 2]
```diff
@@ -269,6 +281,30 @@ public void shouldListAllAccounts() throws Exception {
...
+        databaseTestHelper.addGatewayAccount(accountId.toString(), "provider-1", ImmutableMap.of("user", fuser, "password", "word"), "service-name-1", TEST, "description-1", "analytics-id-1");
+        Optional<GatewayAccountEntity> gatewayAccountOptional = gatewayAccountDao.findById(accountId);
+        assertThat(gatewayAccountOptional.isPresent(), is(true));
+        GatewayAccountEntity gatewayAccountEntity = gatewayAccountOptional.get();
+        assertThat(gatewayAccountEntity.getNotifySettings(), is(nullValue()));
+        Map<String, String> notifySettings = ImmutableMap.of("notify_api_token", notifyAPIToken, "notify_template_id", notifyTemplateId);
+        gatewayAccountEntity.setNotifySettings(notifySettings);
+        gatewayAccountDao.merge(gatewayAccountEntity);
+
+        gatewayAccountOptional = gatewayAccountDao.findById(accountId);
```
Comment (kakumara) [inline src/test/java/uk/gov/pay/connector/it/dao/GatewayAccountDaoITest.java]: Suggestion: This test is okay, but a bit verbose and sometime have a chance of fake success due to the way you test and assert both using JPA. Note JPA entity manager caches entities and though it looks like it reads the saved values from DB it actually gives you back the cached updated entity (which is not in DB yet)
I would have used JPA (DAO) to save the notify_settings and used DatabaseTestHelper to assert if the things got save in the DB. That bypasses JPA and goes directly to database .

File: src/test/java/uk/gov/pay/connector/model/domain/JsonDBFieldConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import org.junit.Test;
+import org.postgresql.util.PGobject;
+
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+public class JsonDBFieldConverterTest {
```
Comment (kakumara) [inline src/test/java/uk/gov/pay/connector/model/domain/JsonDBFieldConverterTest.java]: 👍

**Review by kakumara — COMMENTED** [review]
other than that 👍

File: src/test/java/uk/gov/pay/connector/it/dao/GatewayAccountDaoITest.java (hunk 2 above)
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/it/dao/GatewayAccountDaoITest.java]: 👍

**Review by kakumara — DISMISSED** [review]
LGTM

**Review by kakumara — DISMISSED** [review]
LGTM

**Review by kakumara — APPROVED** [review]
LGTM

**Pull request merged**